ArtOfIllusion/src/artofillusion/image/SVGImage.java

```diff
@@ -373,35 +399,100 @@ public void getGradient(Vec2 grad, int component, boolean wrapx, boolean wrapy,
     }
   }
 
-  /** Get a scaled down copy of the image, to use for previews.  This Image will be no larger
-      (but may be smaller) than PREVIEW_WIDTH by PREVIEW_HEIGHT. */
+  /** Get a scaled down copy of the image, to use for previews.  The dimensions of the 
+      Image will be no larger but may be smaller than  PREVIEW_WIDTH by PREVIEW_HEIGHT. */
+
+  @Override
+  public Image getMapImage(int size)
+  {
+    return getPreview(size);
+  }
 
   @Override
   public Image getPreview()
   {
-    return preview;
+    return getPreview(PREVIEW_SIZE_DEFAULT);
+  }
+
+  /** Get a scaled down copy of the image, to use for previews.  The dimensions of the 
+      Image will be no larger but may be smaller than size. */
+
+  @Override
+  public Image getPreview(int size)
```

Comment:
Here again you're accessing the soft reference in a dangerous way.  Everywhere you have `return preview.get()` you should make sure it isn't null.